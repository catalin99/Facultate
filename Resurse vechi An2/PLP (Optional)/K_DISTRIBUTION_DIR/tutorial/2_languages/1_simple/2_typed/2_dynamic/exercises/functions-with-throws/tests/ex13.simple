int inc(int x) {
  return x+1;
}

void f(int x) throws int, string, (int->int) {
  try {
    if(x==0) {
      throw -1;
    } else {
      if(x==1) {
        throw false;
      } else {
        if(x==2) {
          throw "error";
        } else {
          throw inc;
        }
      }
    }
  } catch(bool b) {
    throw "false";
  }
}

void main() {
  try {
    f(0);                   // rejected by type checker
  } catch(int x) {
    print(x,"\n");
  }
  try {
    f(1);                   // rejected by type checker
  } catch(string x) {
    print(x,"\n");
  }
  try {
    f(2);                   // rejected by type checker
  } catch(string x) {
    print(x,"\n");
  }
  try {
    f(3);                   // rejected by type checker
  } catch(int->int x) {
    print(x(7),"\n");
  }
}
