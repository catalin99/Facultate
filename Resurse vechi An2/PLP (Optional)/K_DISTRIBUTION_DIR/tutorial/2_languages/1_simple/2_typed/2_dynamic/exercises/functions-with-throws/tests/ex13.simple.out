-1
false
error
8
